ersilia/core/tracking.py

```diff
@@ -275,53 +272,57 @@ def upload_to_cddvault(output_df, api_key):
         return False
 
 
-
 def read_csv(file_path):
     """
     Reads a CSV file and returns the data as a list of dictionaries.
 
     :param file_path: Path to the CSV file.
     :return: A list of dictionaries containing the CSV data.
     """
-    with open(file_path, mode='r') as file:
+
+    with open("output.csv", "w", newline="") as csvfile:
```

Comment:
Maybe we can write this file outside either directly in the track function or have a new function called something like "write_result_csv_when_not_a_file". Right now this function is misleading for two reasons:
1. It's writing inside when it's supposed to only read csvs
2. Every time the function is invoked, it will create a file called 'output.csv' even when an input file (or input) is passed.

Reply:
I have made changes to this, the csv file is written inside the track function if result is a generator object